OrbitGl/TracepointTrack.cpp

```diff
@@ -53,13 +69,35 @@ void TracepointTrack::UpdatePrimitives(uint64_t min_tick, uint64_t max_tick, Pic
 
   const Color kWhite(255, 255, 255, 255);
 
-  for (auto it = tracepoints.lower_bound(min_tick); it != tracepoints.upper_bound(max_tick); ++it) {
-    uint64_t time = it->first;
-    if (time < max_tick) {
-      Vec2 pos(time_graph_->GetWorldFromTick(time), pos_[1]);
-      batcher->AddVerticalLine(pos, -track_height, z, kWhite);
-    } else {
-      return;
+  const Color kGreenSelection(0, 255, 0, 255);
+
+  if (!picking) {
+    for (auto it = tracepoints.lower_bound(min_tick); it != tracepoints.upper_bound(max_tick);
+         ++it) {
+      uint64_t time = it->first;
+      if (time < max_tick) {
+        Vec2 pos(time_graph_->GetWorldFromTick(time), pos_[1]);
+        batcher->AddVerticalLine(pos, -track_height, z, kWhite);
+      } else {
+        return;
+      }
+    }
+  } else {
+    constexpr const float kPickingBoxWidth = 9.0f;
+    constexpr const float kPickingBoxOffset = (kPickingBoxWidth - 1.0f) / 2.0f;
+
+    for (auto it = tracepoints.lower_bound(min_tick); it != tracepoints.upper_bound(max_tick);
+         ++it) {
+      uint64_t time = it->first;
+      if (time > min_tick && time < max_tick) {
```

Comment:
Similar to the above - `time` is between `min_tick` and `max_tick`; the `if` is unneeded.

Reply:
Thanks!